# Non Graded Challenge 5

# Linear Algebra (Eigen Decomposition)

Nama : Raditya Saputra

Batch : 007 FTDS

# Library yang digunakan numpy

In [2]:
import numpy as np
from numpy.linalg import eig

# Soal nomor 1

- Apakah $\begin{bmatrix} 1 \ 3 \end{bmatrix}$ adalah eigenvector dari $\begin{bmatrix} 1 \ -1 \ 6 \ 4 \end{bmatrix}$ Jika iya, berapakah eigenvalue-nya?

In [3]:
A = np.array ([[1, -1], [6, 4]])

In [4]:
A

array([[ 1, -1],
       [ 6,  4]])

In [5]:
A.shape

(2, 2)

# Eigen Values dan Eigen Vector

In [6]:
values, vectors = eig (A)

# Eigen Values = 

In [8]:
values

array([2.5+1.93649167j, 2.5-1.93649167j])

# Eigen Vectors =

In [19]:
vectors # vectors satuan/unit

array([[-0.23145502+0.29880715j, -0.23145502-0.29880715j],
       [ 0.9258201 +0.j        ,  0.9258201 -0.j        ]])

# Pembuktiaan eigen vectors pada matriks :
dilakukan dengan melakukan perhitungan jarak/panjang vectors

In [16]:
eig_vec 

array([[1],
       [3]])

In [20]:
A.shape

(2, 2)

In [22]:
eig_vec.shape

(2, 1)

Perkalian antara matriks dengan vectors

In [23]:
hasil_perkalian = A.dot(eig_vec)

In [24]:
hasil_perkalian

array([[-2],
       [18]])

Norm

In [32]:
a = -2**2
b = 18**2
print (a+b)


320


In [34]:
np.sqrt (320)

17.88854381999832

In [35]:
np.array ([-2, 18]) / np.sqrt (320)

array([-0.1118034 ,  1.00623059])

# Hasil pembuktian

([-0.1118034 ,  1.00623059]) != [ 0.9258201 +0.j        ,  0.9258201 -0.j        ]])

# Overall Analysis

- apakah eigen vector ([[1],[3]]) dari matriks ([[1, -1], [6, 4]]) : tidak

- eigen values : ([2.5+1.93649167j, 2.5-1.93649167j]) memiliki bilangan imajiner, apabila eigen values berisikan bilangan imajiner, berarti matriks tersebut tidak ada eigen vectors.